src/main/java/org/wso2/carbon/connector/ISO8583/ISO8583MessageProducer.java

```diff
@@ -0,0 +1,91 @@
+import org.jpos.iso.ISOException;
+import org.jpos.iso.ISOMsg;
+import org.jpos.iso.ISOPackager;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+
+import javax.xml.namespace.QName;
+import java.util.Iterator;
+
+/**
+ * Class for get host, port and iso fields from messageContext
+ */
+public class ISO8583MessageProducer extends AbstractConnector {
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        try {
+            String host = (String) messageContext.getProperty(ISO8583Constant.HOST);
+            int port = Integer.parseInt((String) messageContext.getProperty(ISO8583Constant.PORT));
+            packedISO8583Message(messageContext, host, port);
+        } catch (NumberFormatException e) {
+            handleException("The port number does not contain a parsable integer", e, messageContext);
+        }
+    }
+
+    /**
+     * packed the fields to get String of isoMessage
+     * @param host  the localhost
+     * @param port  5010
+     * @param msgContext the message context
+     */
+    public void packedISO8583Message(MessageContext msgContext, String host, int port) {
+        try {
+            ISOPackager packager = ISO8583PackagerFactory.getPackager();
+            SOAPEnvelope soapEnvelope = msgContext.getEnvelope();
+            OMElement getElements = soapEnvelope.getBody().getFirstElement();
+            ISOMsg isoMsg = new ISOMsg();
+            isoMsg.setPackager(packager);
+            Iterator fields = getElements.getFirstChildWithName(
+                    new QName(ISO8583Constant.TAG_DATA)).getChildrenWithLocalName(ISO8583Constant.TAG_FIELD);
+            while (fields.hasNext()) {
+                OMElement element = (OMElement) fields.next();
+                String getValue = element.getText();
+                try {
+                    int fieldID = Integer.parseInt(element.getAttribute(
+                            new QName(ISO8583Constant.TAG_ID)).getAttributeValue());
+                    isoMsg.set(fieldID, getValue);
+                } catch (NumberFormatException e) {
+                    log.error("The fieldID does not contain a parsable integer" + e.getMessage(), e);
+                }
+            }
+            byte[] data = isoMsg.pack();
+            String packedMessage = new String(data).toUpperCase();
+            handleMessage(msgContext, packedMessage, host, port);
+        } catch (ISOException e) {
+            handleException("Couldn't packed ISO8583 Messages", e, msgContext);
+        }
+    }
+
+    /**
+     * handle the isoMessage with Test Server
+     *
+     * @param isoMessage String of packed Message
+     * @param host  the localhost
+     * @param port  establish connection with port 5010
+     * @param msgContext the message context
+     */
+
+    public void handleMessage(MessageContext msgContext, String isoMessage, String host, int port) {
```

Comment:
No need to have a separate method just for single line. Better have a static method.